public/docs/ts/latest/guide/testing.jade

```diff
@@ -1,102 +1,1758 @@
+  a little at a time. Read the early part of this chapter first
+  to get the basics before trying to absorb the full API.
+
 .alert.is-important
   :marked
-    The testing chapter is still under development.
-    Please bear with us as we both update and complete it.
+    The _TestBed_ is officially _experimental_ and thus subject to change.
+    Consult the [API reference](../api/core/testing/index/TestBed-class.html) for the latest status.
+
+:marked
+  The module definition passed to `configureTestingModule`, 
+  is a subset of the `@NgModule` metadata properties.
+code-example(format="." language="javascript").
+  type TestModuleMetadata = {
+    providers?: any[];
+    declarations?: any[];
+    imports?: any[];
+    schemas?: Array&lt;SchemaMetadata | any[]&gt;;
+  };
+
+:marked
+  Each overide method takes a `MetadataOverride<T>` where `T` is the kind of metadata
+  appropriate to the method, the parameter of an `@NgModule`, `@Component`, `@Directive`, or `@Pipe`.
+
+code-example(format="." language="javascript").
+  type MetadataOverride<T> = {
+    add?: T;
+    remove?: T;
+    set?: T;
+  };
+
+:marked
+a#testbed-methods
+:marked
+  The `TestBed` API consists of static class methods that either update or reference a _global_ instance of the`TestBed`.
+  
+  Internally, all static methods cover methods of the current runtime TestBed instance that is also returned by the `getTestBed()` function.
+
+  Call `TestBed` methods _within_ a `BeforeEach()` to ensure a fresh start before each individual test.
+
+  Here are the most important static methods, in order of likely utility.
+table
+  tr
+    th Methods
+    th Description
+  tr
+    td(style="vertical-align: top") <code>configureTestingModule</code>
+    td
+      :marked
+        The testing shims (`karma-test-shim`, `browser-test-shim`) 
+        establish the [initial test environment](#a#testbed-initTestEnvironment) and a default test module.
+        The default test module is configured with basic declaratives and some Angular service substitutes (e.g. `DebugDomRender`) 
+        that every tester needs.
+        
+        Call `configureTestingModule` to refine the test module configuration for a particular set of tests
+        by adding and removing imports, declarations (of components, directives, and pipes), and providers.
+  tr
+    td(style="vertical-align: top") <code>compileComponents</code>
+    td
+      :marked
+        Compile the test module asynchronously after you've finished configuring it.
+        You **must** call this method if _any_ of the test module components have a `templateUrl`
+        or `styleUrls` because fetching component template and style files is necessarily asynchronous.
+        See [above](#compile-components).
+
+        Once called, the `TestBed` configuration is frozen for the duration of the current spec.
+
+  tr
+    td(style="vertical-align: top") <code>createComponent<T></code>
+    td
+      :marked
+        Create an insta
```

Comment:
I don't see where "your" is needed. Are you objecting to "on the current.."? I don't see anything wrong with that.